miprometheus/problems/seq_to_seq/VQA/cog/cog.py

```diff
@@ -0,0 +1,381 @@
+ #!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) IBM Corporation 2018
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""cog.py: Implementation of Google's COG dataset. https://arxiv.org/abs/1803.06092"""
+
+__author__ = "Emre Sevgen"
+
+import torch
+import gzip
+import json
+import os
+import numpy as np
+
+from miprometheus.utils.data_dict import DataDict
+from miprometheus.problems.seq_to_seq.VQA.VQA_problem import VQAProblem
+from miprometheus.problems.seq_to_seq.VQA.cog.cog_utils import json_to_img as jti
+
+class COGDataset(VQAProblem):
+	"""
+	The COG dataset is a sequential VQA dataset. Inputs are a sequence of images of simple shapes and characters on a black \
+ 	background, and a question based on these objects that relies on memory which has to be answered at every step of the \
+	sequence.
+
+	"""
+
+	def __init__(self, params):
+		"""
+		Initializes the COG Dataset problem:
+
+			- Calls ``problems.problem.VideoTextToClassProblem`` class constructor,
+			- Sets the following attributes using the provided ``params``:
+
+				- ``self.root_folder`` (`string`) : Root directory of dataset where ``processed/training.pt``\
+					and ``processed/test.pt`` will be saved,
+				- ``self.data_folder`` (`string`) : Data directory where dataset is stored. If using canonical \
+									or hard dataset, simply point to 'data_X_Y_Z' folder.
+				- ``self.set`` (`string`) : 'val', 'test', or 'train'
+				- ``self.tasks`` (`string or list of string`) : Which tasks to use. 'class', 'reg', 'all', or a 
+\ list of tasks such as ['AndCompareColor', 'AndCompareShape']. Only selected tasks will be used.
+				- ``self.dataset_type`` (`string`) : Which dataset to use, 'canonical', 'hard', or \
+								'generated'. If 'generated', please specify 'sequence_length', \
+								'memory_length', and 'max_distractors'.
+
+		:param params: Dictionary of parameters (read from configuration ``.yaml`` file).
+
+		"""
+	
+		# Call base class constructors
+		super(COGDataset, self).__init__(params)
+
+		# Set default parameters.
+		self.params.add_default_params({'root_folder': '~/data/COG', 'data_folder': '~/data/COG', 'set': 'train', 
+'tasks': 'class', 'dataset_type': 'canonical'})
+
+		# Retrieve parameters from the dictionary
+		self.root_folder= params['root_folder']
```

Comment:
You can use `os.path.expanduser(params['root_folder'])` to safely handle relative & absolute paths